internal/common/logs.go

```diff
@@ -50,15 +53,19 @@ func writeToLogs(filePath string, duration time.Duration, logLinesPerMinute int)
 	defer ticker.Stop()
 	endTimeout := time.After(duration)
 
-	//loop until the test duration is reached
+	// Sending the logs within the first minute before the ticker kicks in the next minute
+	for i := 0; i < logLinesPerMinute; i++ {
+		_, err := f.WriteString(fmt.Sprintf(logLine, i))
+		if err != nil {
+			return err
+		}
+	}
+
 	for {
 		select {
 		case <-ticker.C:
 			for i := 0; i < logLinesPerMinute; i++ {
-				_, err := f.WriteString(fmt.Sprintf("# %d - This is a log line.", i))
-				if err != nil {
-					log.Printf("Error in writing a string to the file %s: %v ", filePath, err)
-				}
+				f.WriteString(fmt.Sprintf(logLine, i))
```

Comment:
It is not a guarantee that just because the first write succeeded every other write will also succeed.
That said, I think it is fine for "test code" to have a bit less error checking.
Another option is to use "assert" or "panic" on any unexpected failure (no need to gracefully handle/recover from errors, just fail loudly).